### ISSUE TITLE
Create release checklist

### ISSUE THREAD

**Issue by letmaik** [microsoft/scitt-ccf-ledger#51]
This should be a markdown file that contains steps that must happen before cutting a new release.

Something like:

-------------

Copy the following list into a new issue named "<x.y.z> release checklist".
Go through each item, tick it off and once everything is done close the issue.

- [ ] If the CCF version has been upgraded, read all relevant CCF changelog entries and create issues for any outstanding tasks
- [ ] Make sure all issues/PRs for the current [milestone](https://github.com/microsoft/scitt-ccf-ledger/milestones) are closed/merged
- [ ] Create a PR to update `CHANGELOG.md`
- [ ] Create a [release](https://github.com/microsoft/scitt-ccf-ledger/releases) with the new version number as git tag
- [ ] Check that CI is green
